pybossa/jobs.py

```diff
@@ -385,10 +387,11 @@ def get_autoimport_jobs(queue='low'):
 @with_cache_disabled
 def get_project_stats(_id, short_name):  # pragma: no cover
     """Get stats for project."""
-    import pybossa.cache.project_stats as stats
+    with current_app.request_context(DUMMY_ENVIRON):
```

Comment:
Why do we want to request dummy context instead of the real context?

Reply:
> 

It's a job so it's not providing an API and there is not real HTTP connection. But, update_stats-> stats_format_dates-> gettext->...(some flask lib) -> _get_locale, and we use "request" object in the [_get_locale](https://github.com/bloomberg/pybossa/blob/0b52645f618661ead7ff0d869218bd9a9cfe9e25/pybossa/core.py#L340). It is a workaround.

Reply:
Is it more standard to create a context instead? 
```
app = create_app(run_as_server=False)
```
and do similar thing
```
with app.app_context():
       # do something.
```

Reply:
> 

If you check the commit history of this pr, you will find my first [commit](https://github.com/bloomberg/pybossa/pull/678/commits/1dfc0e94b963b013eeb6516accb69e7a47442aec) was trying to change in this way. It was proved not working in QA.

Reply:
Do you know why it wouldn't work? it seems to be working in other code, e.g. https://github.com/bloomberg/pybossa/blob/e3114b3f719575aec9920cacfba3ebc9e619ca5e/cli.py#L151

Reply:
If this works it is probably because none of the function finally calls any function using "request" object.